### DISCUSSION TITLE
Read-S3Object error:  "Could not load type 'Amazon.S3.ChecksumMode' from assembly 'AWSSDK.S3, Version=3.3.0.0, Culture=neutral, PublicKeyToken=xxxx'."

### DISCUSSION THREAD

**Discussion by ghost** [aws/aws-tools-for-powershell#309] [General]
Hi,

We've been using AWSPowershell module to download files from a Wasabi bucket. On some devices, we've gotten an error when calling "Read-S3Object":

`Could not load type 'Amazon.S3.ChecksumMode' from assembly 'AWSSDK.S3, Version=3.3.0.0, Culture=neutral, PublicKeyToken=xxxx'.`

On one device with this issue, I also tried running "Get-S3Object" and it errored with:

`Get-S3Object : Method not found: 'Amazon.Runtime.IAWSTokenProvider Amazon.Runtime.ClientConfig.get_AWSTokenProvider()'.`

It's only some computers with these issues and I can't seem to find the smoking gun, whether it's a bug or if there's some common dependency that's missing.

Anyone have any insight what could be causing that?

Thanks!

**Comment by ghost** [comment]
Bump?

I've tried uninstall the PS module and reinstalling, I've tried updating .NET and C++ redistributables, I've tried disabling security tools, I've tried a ton of things and I can't seem to fix the issue once it's starts on a device.

**Comment by ashishdhingra** [comment]
@ghost Kindly try uninstalling AWS Tools module and re-install them. You may use `-Cleanup` parameter of `Install-AWSToolsModule`.

@github-actions proposed-answer
